### ISSUE TITLE
CMake enhancement

### ISSUE THREAD

**Issue by PaulHariel** [luccanunes/discord.cpp#25]
Can I add CMake to your project? It will be easier to use for both windows and linux users

**Comment by luccanunes** [comment]
You sure can. I've been studying CMake lately because I'm not really great with it and have also been working on a cross-platform documentation, using VSCode, vcpkg and CMake. Any contributions are very welcome!

**Comment by PaulHariel** [comment]
alrighttt i'll try to add it when i have the time
And cross platform documentation like doxygen ?

**Comment by luccanunes** [comment]
No, I mean like a documentation with the set up steps for cross-platform development

**Comment by PaulHariel** [comment]
oh okay

**Comment by ghost** [comment]
Could premake possibly be included? If not could I attempt to include it?

**Comment by luccanunes** [comment]
I'm not familiar with premake, so I can't really talk about it.
But if it's a good thing that makes the set up easier and doesn't conflict with anything else, please do!
All contributions are more then welcome!

**Comment by PaulHariel** [comment]
well, the cmake looks kinda broken on windows, i'll rework that

**Comment by luccanunes** [comment]
No worries, take your time

**Comment by luccanunes** [comment]
@JacqueSatan Would help me getting started with the CMake setup? As I mentioned, I'm not very good at it and keep getting errors when I trying using the CMake version. I only tested it on my raspberry pi runing Raspbian so far.

**Comment by PaulHariel** [comment]
yeah Jacques#3352 on discord
